sway-core/src/type_system/unify/unifier.rs

```diff
@@ -436,7 +446,7 @@ impl<'a> Unifier<'a> {
         span: &Span,
         received_type_argument: &TypeArgument,
         expected_type_argument: &TypeArgument,
-    ) {
+    ) -> bool {
```

Comment:
I'm not a fan of using bool as a return type when it's ambiguous what that could mean.

In this case it seems pretty natural, but at the least the behavior should be documented in the function's doccomment.

Reply:
Done.